crates/sui-core/src/transaction_orchestrator.rs

```diff
@@ -397,73 +432,102 @@ where
-            total_req_received.with_label_values(&["wait_for_effects_cert"]);
-        let total_req_received_wait_for_local_execution =
-            total_req_received.with_label_values(&["wait_for_local_execution"]);
+        let total_req_received_single_writer =
+            total_req_received.with_label_values(&[TX_TYPE_SINGLE_WRITER_TX]);
+        let total_req_received_shared_object =
+            total_req_received.with_label_values(&[TX_TYPE_SHARED_OBJ_TX]);
+        let total_req_received_staking = total_req_received.with_label_values(&[TX_TYPE_STAKING]);
 
         let good_response = register_int_counter_vec_with_registry!(
             "tx_orchestrator_good_response",
-            "Total number of good responses Transaction Orchestrator generates, group by request type",
-            &["request_type"],
+            "Total number of good responses Transaction Orchestrator generates, group by tx type",
+            &["tx_type"],
             registry
         )
         .unwrap();
 
-        let good_response_wait_for_effects_cert =
-            good_response.with_label_values(&["wait_for_effects_cert"]);
-        let good_response_wait_for_local_execution =
-            good_response.with_label_values(&["wait_for_local_execution"]);
+        let good_response_single_writer =
+            good_response.with_label_values(&[TX_TYPE_SINGLE_WRITER_TX]);
+        let good_response_shared_object = good_response.with_label_values(&[TX_TYPE_SHARED_OBJ_TX]);
+        let good_response_staking = good_response.with_label_values(&[TX_TYPE_STAKING]);
 
         let req_in_flight = register_int_gauge_vec_with_registry!(
             "tx_orchestrator_req_in_flight",
-            "Number of requests in flights Transaction Orchestrator processes, group by request type",
-            &["request_type"],
+            "Number of requests in flights Transaction Orchestrator processes, group by tx type",
+            &["tx_type"],
             registry
         )
         .unwrap();
 
-        let req_in_flight_wait_for_effects_cert =
-            req_in_flight.with_label_values(&["wait_for_effects_cert"]);
-        let req_in_flight_wait_for_local_execution =
-            req_in_flight.with_label_values(&["wait_for_local_execution"]);
+        let req_in_flight_single_writer =
+            req_in_flight.with_label_values(&[TX_TYPE_SINGLE_WRITER_TX]);
+        let req_in_flight_shared_object = req_in_flight.with_label_values(&[TX_TYPE_SHARED_OBJ_TX]);
+
+        let request_latency = HistogramVec::new_in_registry(
+            "tx_orchestrator_request_latency",
+            "Time spent in processing one Transaction Orchestrator request",
+            &["type"],
+            registry,
+        );
+        let wait_for_finality_latency = HistogramVec::new_in_registry(
+            "tx_orchestrator_wait_for_finality_latency",
+            "Time spent in waiting for one Transaction Orchestrator request gets finalized",
+            &["type"],
```

Comment:
Actually there's other code that might touch the system object, frenemies being one of them. Third party can also write Move functions that take in the system state object. Maybe we can add a label for the calling module of the transaction? This way we can see what's driving the high demand for system state obj: whether it's frenemies game, staking functions or someone else's module.

Reply:
Is staking/unstaking/switch gonna dominate txes that call 0x5? I don't think we should log every move call if we don't care about them. Another way is to log only staking related function calls, assuming there are only a few.